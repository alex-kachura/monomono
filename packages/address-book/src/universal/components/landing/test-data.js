export const TESTING_DATA = {
  'primary-addresses': {
    clubcard: {
      label: 'Clubcard Address',
      addressLines: ['Lever Building', '85 Clerkenwell Road', 'Holborn'],
      postTown: 'London',
      postCode: 'EC1R 5AR',
      telephoneNumbers: [
        {
          label: 'Day',
          value: '123456789',
        },
      ],
    },
    grocery: {
      label: 'Home',
      addressLines: ['Tesco Pear Building', 'Shire Park', 'Kestrel Way'],
      postTown: 'Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch',
      postCode: 'AA9A 9AA',
      telephoneNumbers: [
        {
          label: 'Day',
          value: '123456789',
        },
        {
          label: 'Evening',
          value: '123456789',
        },
        {
          label: 'Mobile',
          value: '123456789',
        },
      ],
    },
  },
  'other-addresses': [
    {
      label: "My favourite son's uni flat",
      addressLines: ['122', 'Reservoir Road', 'Birmingham'],
      postTown: 'Erdington',
      postCode: 'B23 6DN',
      telephoneNumbers: [
        {
          label: 'Day',
          value: '123456789',
        },
        {
          label: 'Evening',
          value: '123456789',
        },
        {
          label: 'Mobile',
          value: '123456789',
        },
      ],
    },
    {
      label: "My other son's uni flat",
      addressLines: ['122', 'Reservoir Road', 'Birmingham'],
      postTown: 'Erdington',
      postCode: 'B23 6DN',
      telephoneNumbers: [
        {
          label: 'Day',
          value: '123456789',
        },
        {
          label: 'Evening',
          value: '123456789',
        },
        {
          label: 'Mobile',
          value: '123456789',
        },
      ],
    },
    {
      label: 'Home 2',
      addressLines: ['Tesco Pear Building', 'Shire Park', 'Kestrel Way'],
      postTown: 'Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch',
      postCode: 'AA9A 9AA',
      telephoneNumbers: [
        {
          label: 'Day',
          value: '123456789',
        },
        {
          label: 'Evening',
          value: '123456789',
        },
        {
          label: 'Mobile',
          value: '123456789',
        },
      ],
    },
  ],
};
