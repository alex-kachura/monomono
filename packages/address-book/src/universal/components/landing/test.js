import React from 'react';
import { render } from 'react-testing-library';
import { renderProviders } from '@oneaccount/react-foundations/lib/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { DefaultThemeProvider } from '@beans/theme';
import LandingPage from './';
import { mockData } from './test-data';

describe('[Component: LandingPage]', () => {
  let component;
  const mockLocation = {
    pathname: 'mock-path',
    search: 'mock-search',
  };

  beforeEach(() => {
    const providers = renderProviders({
      appConfig: {
        getLocalePhrase: (key) => key,
        rootPath: '/mock/math',
      },
      initialPageData: {
        addresses: mockData,
        banner: {
          bannerType: 'error',
          title: '',
          description: '',
        },
      },
      children: (
        <DefaultThemeProvider>
          <BrowserRouter>
            <LandingPage location={mockLocation} />
          </BrowserRouter>
        </DefaultThemeProvider>
      ),
    });

    component = render(providers);
  });

  it('should render correctly', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
