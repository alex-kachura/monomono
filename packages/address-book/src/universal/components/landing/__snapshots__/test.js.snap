// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: LandingPage] should render correctly 1`] = `
<DocumentFragment>
  <h1
    class="base-components__BaseElement-sc-150tvch-0 sc-jKJlTe kaiMDg typography___styledElement-sc-186l0ce-0 hmbiuS"
  >
    pages.landing.title
  </h1>
  <div
    class="sc-kpOJdX gNYMhr"
    style="background-color: rgb(251, 239, 239);"
  >
    <div
      class="sc-dxgOiQ iXUGkK"
    >
      <div
        class="styled__Container-sc-1bf1p22-0 fRWwDM"
      >
        <svg
          aria-hidden="true"
          class="styled__StyledSVG-sc-1nulmu6-0 fnQPMZ"
          color="currentColor"
          height="21"
          preserveAspectRatio="xMinYMax meet"
          viewBox="0 0 20 21"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 15V1.5c2.623 0 4.184.43 6.311 2.106M3.5 11.5c-1.103 0-2-.897-2-2s.897-2 2-2 2 .897 2 2-.897 2-2 2zM16.5
        11.5c-1.103 0-2-.897-2-2s.897-2 2-2 2 .897 2 2-.897 2-2 2zM16.742 20.5c-2.037-1.173-4.609-2-6.742-2-1.879
        0-4.104.638-6 1.591-.257.129-.5.269-.743.409"
            fill="none"
            stroke="#ffffff"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-ckVGcZ gcUxmH"
    >
      <h5
        class="base-components__BaseElement-sc-150tvch-0 typography___styledElement-sc-186l0ce-0 fREkql"
      />
    </div>
  </div>
  <div
    class="sc-eNQAEJ gwllH styled-sc-140lqfg-0 hKAvLp"
  >
    <h2
      class="base-components__BaseElement-sc-150tvch-0 sc-kkGfuU bZEveT typography___styledElement-sc-186l0ce-0 iswQsi"
    >
      pages.landing.primary-address.label
    </h2>
    <div
      class="sc-hMqMXs eWZRIv styled-cgoahb-0 fFoLXC"
    >
      <div
        class="sc-ifAKCX jmlQG styled-tbp4qh-0 hbPPrO"
      >
        <div
          class="sc-bxivhb eyMKcU"
          id="edit-address-clubcard"
        >
          <div
            class="primary-address__address-header sc-EHOje jrvcQh"
          >
            <svg
              aria-hidden="true"
              class="sc-bZQynM eBMLPR styled__Container-sc-1bf1p22-0 gJOjrx"
              color="currentColor"
              height="18"
              preserveAspectRatio="xMinYMax meet"
              viewBox="0 0 24 18"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 16.5H3c-.8 0-1.5-.7-1.5-1.5V3c0-.8.7-1.5 1.5-1.5h18c.8 0 1.5.7 1.5 1.5v12c0 .8-.7 1.5-1.5 1.5zM10
        7.5h4.1M8 10.5h8"
                fill="none"
                stroke="#00539f"
              />
            </svg>
            <div
              class="sc-dnqmqq kWHYmu"
            >
              <h5
                class="base-components__BaseElement-sc-150tvch-0 typography___styledElement-sc-186l0ce-0 fREkql"
              >
                pages.landing.primary-address.header.clubcard
              </h5>
            </div>
          </div>
          <div
            class="sc-bdVaJa efQDjt"
          >
            <h4
              class="base-components__BaseElement-sc-150tvch-0 nickname sc-htpNat ctyXjV typography___styledElement-sc-186l0ce-0 ixZPNc"
            >
              Clubcard
            </h4>
            <p
              class="base-components__BaseElement-sc-150tvch-0 address-line-0 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              85 Clerkenwell Road
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 town sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              LONDON
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 postcode sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              EC1R 5AR
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 telephone-number sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              pages.landing.telephone.phone: 07345234636
            </p>
          </div>
          <div
            class="sc-gzVnrw jhTHcp"
          >
            <span
              class="edit-address-link-clubcard styled__StandaloneContainer-sc-1xizymv-2 TEXwR"
            >
              <a
                href="/mock/math/edit-clubcard-address?id=NGC_0"
              >
                <span
                  class="styled__StandaloneSpan-sc-1xizymv-1 kqERAn"
                >
                  <span
                    class="beans-link__text styled__TextSpan-sc-1xizymv-3 hWdmzc"
                  >
                    pages.landing.edit-btn
                  </span>
                </span>
              </a>
            </span>
            <p
              class="base-components__BaseElement-sc-150tvch-0 sc-htoDjs dRDBok typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              pages.landing.primary-address.additional.clubcard
            </p>
          </div>
        </div>
      </div>
      <div
        class="sc-ifAKCX jmlQG styled-tbp4qh-0 hbPPrO"
      >
        <div
          class="sc-bxivhb eyMKcU"
          id="edit-address-grocery"
        >
          <div
            class="primary-address__address-header sc-EHOje jrvcQh"
          >
            <svg
              aria-hidden="true"
              class="sc-bZQynM eBMLPR styled__Container-sc-1bf1p22-0 gJOjrx"
              color="currentColor"
              height="22"
              preserveAspectRatio="xMinYMax meet"
              viewBox="0 1 24 22"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 20.5H4l-2.5-11h21zM12 12v6M16 12v6M8 12v6M12 9.5L18 1"
                fill="none"
                stroke="#00539f"
              />
            </svg>
            <div
              class="sc-dnqmqq kWHYmu"
            >
              <h5
                class="base-components__BaseElement-sc-150tvch-0 typography___styledElement-sc-186l0ce-0 fREkql"
              >
                pages.landing.primary-address.header.grocery
              </h5>
            </div>
          </div>
          <div
            class="sc-bdVaJa efQDjt"
          >
            <h4
              class="base-components__BaseElement-sc-150tvch-0 nickname sc-htpNat ctyXjV typography___styledElement-sc-186l0ce-0 ixZPNc"
            >
              home 3
            </h4>
            <p
              class="base-components__BaseElement-sc-150tvch-0 address-line-0 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              85 Clerkenwell Road
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 town sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              LONDON
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 postcode sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              EC1R 5AR
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 telephone-number-0 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              pages.landing.telephone.day: 07345234636
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 telephone-number-1 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              pages.landing.telephone.evening: 07345234635
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 telephone-number-2 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              pages.landing.telephone.mobile: 07345234637
            </p>
          </div>
          <div
            class="sc-gzVnrw jhTHcp"
          >
            <span
              class="edit-address-link-grocery styled__StandaloneContainer-sc-1xizymv-2 TEXwR"
            >
              <a
                href="/mock/math/edit-delivery-address?id=GHS_11225523"
              >
                <span
                  class="styled__StandaloneSpan-sc-1xizymv-1 kqERAn"
                >
                  <span
                    class="beans-link__text styled__TextSpan-sc-1xizymv-3 hWdmzc"
                  >
                    pages.landing.edit-btn
                  </span>
                </span>
              </a>
            </span>
            <p
              class="base-components__BaseElement-sc-150tvch-0 sc-htoDjs dRDBok typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              pages.landing.primary-address.additional.grocery
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-eNQAEJ RhZmx styled-sc-140lqfg-0 hKAvLp"
  >
    <h2
      class="base-components__BaseElement-sc-150tvch-0 sc-kkGfuU bZEveT typography___styledElement-sc-186l0ce-0 iswQsi"
    >
      pages.landing.other-addresses.label
    </h2>
    <p
      class="base-components__BaseElement-sc-150tvch-0 sc-iAyFgw IEbue typography___styledElement-sc-186l0ce-0 gxVCAq"
    >
      pages.landing.other-addresses.sub-label
    </p>
    <div
      class="other-address sc-kEYyzF hwdaVg styled-cgoahb-0 fFoLXC"
    >
      <div
        class="sc-kAzzGY jLybqU styled-tbp4qh-0 fsXMrV"
      >
        <div
          class="sc-chPdSV eEdLch"
        >
          <span
            class="sc-hSdWYo dhsYtq styled__IconButtonContainerWithText-sc-1r4nugl-2 eDfFjL"
          >
            <a
              href="/mock/math/add-delivery-address"
            >
              <div
                class="sc-cvbbAY gkqdlC"
              >
                <span
                  class="styled__IconButtonSpanWithText-sc-1r4nugl-1 ilogHV"
                  id="add-delivery-address"
                >
                  <span
                    class="beans-link__text styled__TextSpan-sc-1r4nugl-3 fXsDiB"
                  >
                    <h4
                      class="base-components__BaseElement-sc-150tvch-0 sc-eHgmQL eCjgBV typography___styledElement-sc-186l0ce-0 ixZPNc"
                    >
                      Add delivery address
                    </h4>
                  </span>
                  <span
                    class="beans-link__icon styled__IconButtonSpan-sc-1r4nugl-4 bmkyLu"
                  >
                    <span
                      class="styled__IconSpan-sc-1r4nugl-5 kuXyIv"
                    >
                      <svg
                        aria-hidden="true"
                        class="styled__Container-sc-1bf1p22-0 gJOjrx"
                        color="currentColor"
                        height="18"
                        preserveAspectRatio="xMinYMax meet"
                        viewBox="0 0 18 18"
                        width="18"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0 9h18M9 0v18"
                          fill="none"
                          stroke="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
            </a>
          </span>
        </div>
      </div>
      <div
        class="sc-kAzzGY jLybqU styled-tbp4qh-0 fsXMrV"
      >
        <div
          class="sc-chPdSV bXcHcM"
        >
          <div
            class="sc-bdVaJa fOmtxQ"
          >
            <h4
              class="base-components__BaseElement-sc-150tvch-0 nickname sc-htpNat ctyXjV typography___styledElement-sc-186l0ce-0 ixZPNc"
            >
              Luis
            </h4>
            <p
              class="base-components__BaseElement-sc-150tvch-0 address-line-0 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              Flat 1
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 address-line-1 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              The Old School
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 address-line-2 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              146 York Way
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 town sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              LONDON
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 postcode sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              N1 0AE
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 telephone-number-0 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              pages.landing.telephone.day: 07427708482
            </p>
            <p
              class="base-components__BaseElement-sc-150tvch-0 telephone-number-1 sc-bwzfXH dihBbR typography___styledElement-sc-186l0ce-0 gxVCAq"
            >
              pages.landing.telephone.evening: 07427708482
            </p>
          </div>
          <div
            class="sc-kgoBCf fsWbQS"
          >
            <span
              class="edit-address-link-grocery styled__StandaloneContainer-sc-1xizymv-2 TEXwR"
            >
              <a
                href="/mock/math/edit-delivery-address?id=GHS_11214795"
              >
                <span
                  class="styled__StandaloneSpan-sc-1xizymv-1 kqERAn"
                >
                  <span
                    class="beans-link__text styled__TextSpan-sc-1xizymv-3 hWdmzc"
                  >
                    pages.landing.edit-btn
                  </span>
                </span>
              </a>
            </span>
            <div
              class="sc-kGXeez gFYsIb"
            />
            <form
              action="/mock/math/"
              method="POST"
            >
              <input
                name="contact-address-id"
                type="hidden"
                value="GHS_11214795"
              />
              <button
                aria-disabled="false"
                class="styled__BaseButton-sc-3mnirm-0 delete-address styled__LinkButton-sc-3mnirm-13 gyNqJj"
                role="button"
                type="submit"
              >
                <span
                  class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
                >
                  <span
                    class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
                  >
                    pages.landing.delete-btn
                  </span>
                </span>
              </button>
              <input
                name="_csrf"
                type="hidden"
                value=""
              />
            </form>
            <div
              class="sc-gqjmRU xUiFG"
              style="transform: translate3d(0, 100%, 0); opacity: 0;"
            >
              <h4
                class="base-components__BaseElement-sc-150tvch-0 sc-VigVT eyiuaP typography___styledElement-sc-186l0ce-0 ixZPNc"
              >
                pages.landing.delete.confirmation-message
              </h4>
              <div
                class="sc-cSHVUG bhIRwB"
              >
                <p
                  class="base-components__BaseElement-sc-150tvch-0 nickname sc-fjdhpX cxEKBW typography___styledElement-sc-186l0ce-0 gxVCAq"
                >
                  Luis
                </p>
                <p
                  class="base-components__BaseElement-sc-150tvch-0 address-line-0 sc-jTzLTM cjflfh typography___styledElement-sc-186l0ce-0 gxVCAq"
                >
                  Flat 1
                </p>
              </div>
              <form
                action="/mock/math/"
                method="POST"
              >
                <input
                  name="contact-address-id"
                  type="hidden"
                  value="GHS_11214795"
                />
                <button
                  aria-disabled="false"
                  class="styled__BaseButton-sc-3mnirm-0 delete-address-confirm sc-jzJRlG cyAuDu styled__PrimaryButton-sc-3mnirm-2 giYBMC"
                  role="button"
                  type="submit"
                >
                  <span
                    class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
                  >
                    <span
                      class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
                    >
                      pages.landing.delete.confirm
                    </span>
                  </span>
                </button>
                <input
                  name="_csrf"
                  type="hidden"
                  value=""
                />
              </form>
              <button
                aria-disabled="false"
                class="styled__BaseButton-sc-3mnirm-0 delete-address-cancel styled__SecondaryButton-sc-3mnirm-3 bzFQfg"
                role="button"
                type="button"
              >
                <span
                  class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
                >
                  <span
                    class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
                  >
                    pages.landing.delete.cancel
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
